core_types/src/entry/mod.rs

```diff
@@ -100,6 +101,13 @@ impl PartialEq for Entry {
 }
 
 impl AddressableContent for Entry {
+    fn address(&self) -> Address {
+        match &self {
+            Entry::AgentId(agent_id) => agent_id.address(),
+            _ => Address::encode_from_str(&String::from(self.content()), Hash::SHA2256),
```

Comment:
hmm, this seems like it should always defer to the `.address()` method of the inner value?

Reply:
Are you sure? Wouldn't that mean that the entry type is not included in the hash? In Montreal we agreed that the "content" that gets hashed needs to include the type.